scripts/prepare.js

```diff
@@ -46,22 +51,30 @@ function downloadIfNeeded(url, dest) {
   if (url.startsWith('file:')) {
     fs.createReadStream(url.substring('file:'.length)).pipe(fs.createWriteStream(dest));
   } else {
-    request(url + '.sha1', { auth }, (error, response, body) => {
+    const callback = (error, response, body) => {
       if (error) {
         throw error;
       } else if (response.statusCode !== HTTP_OK) {
         throw new Error(`Unable to get file ${url}: ${response.statusCode} ${body}`);
       } else {
         downloadIfChecksumMismatch(body, url, dest);
       }
-    });
+    };
+    if (credentialsDefined) {
+      request(url + '.sha1', {auth}, callback);
+    } else {
+      request(url + '.sha1', callback);
+    }
   }
 }
 
 function downloadIfChecksumMismatch(expectedChecksum, url, dest) {
   if (!fs.existsSync(dest)) {
-    request(url, { auth }).pipe(fs.createWriteStream(dest));
-
+    if (credentialsDefined) {
```

Comment:
This check also needs to be performed on line 85, otherwise the download of JAR files fails when credentials are not defined.